pkg/hook/accessrequesthook/accessrequesthook.go

```diff
@@ -166,6 +166,24 @@ func (h Hook) NoEntitlementAccess(ctx context.Context, cfg *config.Context, inpu
 			req.Justification.Reason = &customReason
 		}
 	}
+
+	for i := range req.Entitlements {
+
+		if input.Duration != nil {
+			req.Entitlements[i].Duration = input.Duration
+			continue
+		}
+
+		if result.DurationConfiguration != nil {
+			if result.DurationConfiguration.DefaultDuration != nil {
+				req.Entitlements[i].Duration = result.DurationConfiguration.DefaultDuration
+			} else {
+				req.Entitlements[i].Duration = result.DurationConfiguration.MaxDuration
+			}
+		}
+
+	}
```

Comment:
The duration flag already gets set on the request at the top of the file.
If it is not provided the server chooses the correct duration.
```suggestion
```